(Draft) Revamp JSON-RPC & Typescript generation

Frando opened the pull request:
This PR moves the JSON-RPC and Typescript generation to a new crate [yerpc](https://github.com/Frando/yerpc). It removes quite a lot of code from this crate to focus on the actual Deltachat RPC client. The generated Typescript should be more reliable.

*678 insertions(+), 3330 deletions(-)*

Missing so far: 
- [x] Check if everything works
- [x] Port tests
- [ ] Think about a high-level Deltachat client

fixes #8
fixes #5

Simon-Laux committed:
update to new unrelease 1.70 core
for some reason this breaks the send message test - did something in the contact request logic change?

Simon-Laux committed:
run cargo fmt

Frando committed:
it builds

Frando committed:
improve typescript client

Frando committed:
update packages

Frando committed:
Remove obsolete code

Frando committed:
cargo fmt

Frando committed:
cargo clippy --fix

Simon-Laux commented:
> Think about a high-level Deltachat client

I don't think we need to tackle this point right now, the api is already build to represent what the current Deltachat desktop frontend expects api wise. Sure would be nice to have a higher level api that has pleasant method names and a well thought out structure, but I don't think we need it for this pr, this pr is already a great step forward.

Simon-Laux committed:
make context the first argument

Simon-Laux committed:
Change apis that they take an account ID
closes #8

Simon-Laux committed:
rename `sc` variable to `ctx`
also commit updated `client.ts`

Simon-Laux committed:
add back rust smoke test

Simon-Laux committed:
update example to work again
also expose the client in the js console
to make the example more useful for testing

Simon-Laux committed:
remove unused imports

Simon-Laux committed:
add support for ui config keys
add batchSetConfig
add tests for it
fix the existing tests

Simon-Laux committed:
- add prettier for code formatting
- move dev dependencies to where they belong
- adjust npm scripts

Simon-Laux committed:
fix code formatting

Simon-Laux committed:
enable sourcemap support for tests

Simon-Laux committed:
fix basic test shutdown

Simon-Laux committed:
add server process for test troubleshooting steps

Simon-Laux committed:
disable webserver feature by default again
and instead require it for the bin target

Simon-Laux committed:
fix dependency stuff

Simon-Laux committed:
make getChatlistEntries' listflags optional

Frando committed:
fix Cargo.toml

Simon-Laux committed:
update deps and core to 1.77

Simon-Laux committed:
update typescript-type-def

Frando committed:
update dependencies

Frando committed:
rename to deltachat-jsonrpc

Simon-Laux committed:
add can_send property to chat
closes #5

Frando committed:
cleanup and docs

Frando committed:
fixes

Frando committed:
prettier
